### PR TITLE
coordinator: use history in authority

### PR DESCRIPTION
**Pull request by burgerdev**
This change swaps the in-memory persistence for a filesystem-backed KV store wrapped by `History`.  

Through the use of an internal atomic state pointer, the `Authority` struct is now thread-safe and responds with the correct state (esp. CA certs) even under concurrent calls to `SetManifest`. It also anticipates out-of-band modifications of persistence and syncs accordingly, which should make it ready for a distributed setup.

### CODE REVIEW

File: coordinator/internal/authority/authority.go [hunk 1]
```diff
@@ -181,52 +189,145 @@ func (m *Authority) GetCertBundle(peerPublicKeyHashStr string) (Bundle, error) {
...
-	return m.manifests.All(), m.ca.Load()
+//
+// If no manifest is configured, it returns an empty slice and a nil CA.
+func (m *Authority) GetManifestsAndLatestCA() ([]*manifest.Manifest, *ca.CA, error) {
+	if err := m.sync(); err != nil {
+		return nil, nil, fmt.Errorf("could not sync internal state: %w", err)
+	}
+	state := m.state.Load()
+	if state == nil {
+		return nil, nil, nil
```
Comment (katexochen) [inline coordinator/internal/authority/authority.go:200]: returning nil, nil (,nil) is usually discouraged.

File: coordinator/internal/authority/authority_test.go
Comment (katexochen) [inline coordinator/internal/authority/authority_test.go:1]: Test don't cover many code paths, but better than no tests. :)

**Review by katexochen — CHANGES_REQUESTED** [review]

File: coordinator/internal/authority/authority.go (hunk 1 above)
Comment (burgerdev) [reply coordinator/internal/authority/authority.go:200]: Would you rather fail the call, or return an empty slice and nil CA?

Comment (katexochen) [reply coordinator/internal/authority/authority.go:200]: We are returning an error on the caller side anyway if we get 3x nil, right? So we might want to raise it here.

Comment (burgerdev) [reply coordinator/internal/authority/authority.go]: Added a test for it that revealed a bug :)

File: coordinator/internal/authority/authority_test.go
Comment (burgerdev) [reply coordinator/internal/authority/authority_test.go:1]: I tried to cover some more, but leaving the rest open for a future PR.

**Review by katexochen — APPROVED** [review]

**Comment by katexochen** [comment]
Let's wait with the merge until after v0.7.0.

File: coordinator/internal/authority/authority.go [hunk 2]
```diff
@@ -174,59 +181,160 @@ func (m *Authority) GetCertBundle(peerPublicKeyHashStr string) (Bundle, error) {
...
 
 	return bundle, nil
 }
 
 // GetManifestsAndLatestCA retrieves the manifest history and the currently active CA instance.
-func (m *Authority) GetManifestsAndLatestCA() ([]*manifest.Manifest, *ca.CA) {
-	// TODO(burgerdev): The CA should be tied to the manifest.
-	return m.manifests.All(), m.ca.Load()
+//
+// If no manifest is configured, it returns an empty slice and a nil CA.
```
Comment (burgerdev) [inline coordinator/internal/authority/authority.go:192]: This is not true anymore.

**Pull request merged**